src/app/pages/AccountDetailsPage/ContractCodeCard.tsx

```diff
@@ -0,0 +1,93 @@
+import { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import Card from '@mui/material/Card'
+import CardContent from '@mui/material/CardContent'
+import { ScrollingDiv } from '../../components/PageLayout/ScrollingDiv'
+import { useAccount } from './hook'
+import { useRequiredScopeParam } from '../../hooks/useScopeParam'
+import { useLoaderData } from 'react-router-dom'
+import { CardEmptyState } from './CardEmptyState'
+import Typography from '@mui/material/Typography'
+import { ScrollingDataDisplay } from '../../components/ScrollingDataDisplay'
+import Box from '@mui/material/Box'
+import { CopyToClipboard } from '../../components/CopyToClipboard'
+import { useScreenSize } from '../../hooks/useScreensize'
+
+export const contractCodeContainerId = 'code'
+
+export const ContractCodeCard: FC = () => {
+  const { t } = useTranslation()
+  const scope = useRequiredScopeParam()
+  const address = useLoaderData() as string
+  const { isMobile } = useScreenSize()
+
+  const { isFetched, account } = useAccount(scope, address)
+  const contract = account?.evm_contract
+  const hasCreation = !!contract?.creation_bytecode
+  const hasRuntime = !!contract?.runtime_bytecode
+  const noCode = isFetched && !hasCreation && !hasRuntime
+  return (
+    <Card>
+      <ScrollingDiv id={contractCodeContainerId}>
+        {noCode && <CardEmptyState label={t('contract.noCode')} />}
+        {(hasCreation || hasRuntime) && (
+          <CardContent>
+            {hasCreation && (
+              <>
+                <Box
+                  sx={{
+                    display: 'flex',
+                    justifyContent: 'space-between',
+                    alignItems: 'center',
+                    mb: '10px',
+                  }}
+                >
+                  <Typography variant="h4" component="h4">
+                    {t('contract.creationByteCode')}
+                  </Typography>
+                  <CopyToClipboard
+                    value={contract?.creation_bytecode!}
+                    label={
+                      isMobile
+                        ? t('common.copy')
+                        : t('contract.copyButton', { subject: t('contract.creationByteCode') })
+                    }
+                  />
+                </Box>
+
+                <ScrollingDataDisplay data={contract.creation_bytecode!} />
+              </>
+            )}
+            {hasRuntime && (
+              <>
+                <Box
+                  sx={{
+                    display: 'flex',
+                    justifyContent: 'space-between',
+                    alignItems: 'center',
+                    mt: '10px',
+                    pt: '20px',
```

Comment:
`my: 3` instead of mt+mb
`pt: 4` and it should be only applied only when `hasCreation` is true
Sections for hasCreation and hasRuntime seems the same could be extracted to one component.

Reply:
OK done